From 329433f10d993d69d2f1424d0645cd15d763c9d8 Mon Sep 17 00:00:00 2001
From: Cedric BAIL <cedric.bail@samsung.com>
Date: Wed, 5 Feb 2014 11:10:29 +0900
Subject: [PATCH] eina: restore magic debug option for ABI compatibility
 purpose.

We can't remove that option as it break ABI for people who did turn
it off in 1.7.

- cherry-pick me -
---
 configure.ac | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9ca8f42..4f78ef9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -688,7 +688,6 @@ EFL_LIB_START([Eina])
 
 ### Default values
 
-have_magic_debug="yes"
 have_safety_checks="yes"
 want_log="yes"
 case "${build_profile}" in
@@ -724,7 +723,6 @@ case "${build_profile}" in
 esac
 
 EINA_CONFIG([HAVE_ALLOCA_H], [test "x${ac_cv_working_alloca_h}" = "xyes"])
-EINA_CONFIG([MAGIC_DEBUG], [test "x${have_magic_debug}" = "xyes"])
 EINA_CONFIG([SAFETY_CHECKS], [test "x${have_safety_checks}" = "xyes"])
 EINA_CONFIG([DEFAULT_MEMPOOL], [test "x${want_default_mempool}" = "xyes"])
 
@@ -738,6 +736,18 @@ AC_DEFINE_IF([EINA_STRINGSHARE_USAGE],
    [1], [Report Eina stringshare usage pattern])
 
 ### Additional options to configure
+AC_ARG_ENABLE([magic-debug],
+   [AC_HELP_STRING([--disable-magic-debug], [disable magic debug of eina structure @<:@default=enabled@:>@])],
+   [
+    if test "x${enableval}" = "xyes" ; then
+       have_magic_debug="yes"
+    else
+       have_magic_debug="no"
+    fi
+   ],
+   [have_magic_debug="yes"])
+
+EINA_CONFIG([MAGIC_DEBUG], [test "x${have_magic_debug}" = "xyes"])
 
 ### Checks for programs
 
-- 
1.9.0.rc3

